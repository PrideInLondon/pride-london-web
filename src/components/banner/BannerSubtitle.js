import React from 'react'
import PropTypes from 'prop-types'
import {
  StyledBannerSubtitle,
  BannerSubtitleWrapper,
} from './BannerSubtitle.styles'

BannerSubtitleWrapper.displayName = 'SubtitleWrapper'
StyledBannerSubtitle.displayName = 'SubtitleText'

const BannerSubtitle = ({ children }) => (
  <BannerSubtitleWrapper>
    <StyledBannerSubtitle>{children}</StyledBannerSubtitle>
  </BannerSubtitleWrapper>
)

BannerSubtitle.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
}

export default BannerSubtitle
