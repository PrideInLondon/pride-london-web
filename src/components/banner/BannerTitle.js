import React from 'react'
import PropTypes from 'prop-types'
import { StyledBannerTitle, BannerTitleWrapper } from './BannerTitle.styles'

const BannerTitle = ({ children }) => (
  <BannerTitleWrapper>
    <StyledBannerTitle>{children}</StyledBannerTitle>
  </BannerTitleWrapper>
)

BannerTitle.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
}

export default BannerTitle
