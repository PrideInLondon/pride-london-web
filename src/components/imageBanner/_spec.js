import React from 'react'
import { shallow } from 'enzyme'
import BannerTitle from '../bannerTitle'
import BannerSubtitle from '../bannerSubtitle'
import ImageBanner from './'
import 'jest-styled-components'

describe('ImageBanner', () => {
  it('renders', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a <BannerTitle />', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(
      wrapper
        .find(BannerTitle))
      .toHaveLength(1)
  })

  it('renders a <BannerSubtitle />', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(
      wrapper
        .find(BannerSubtitle))
      .toHaveLength(1)
  })

  it('renders an img if passed an imageSrc prop', () => {
    const wrapper = shallow(<ImageBanner imageSrc="foo" />)
    expect(
      wrapper.
        find('img'))
      .toHaveLength(1)
  })

  it('renders render an img if not passed an imageSrc prop', () => {
    const wrapper = shallow(<ImageBanner />)
    expect(
      wrapper
        .find('img'))
      .toHaveLength(0)
  })

  it('renders an img with imgSrc from props', () => {
    const imageSrc = 'https://www.image.com'
    const wrapper = shallow(<ImageBanner imageSrc={imageSrc} />)
    expect(
      wrapper
        .find('img')
        .props()
        .src)
      .toBe(imageSrc)
  })

  it('renders an img with altText from props', () => {
    const altText = 'background image'
    const wrapper = shallow(<ImageBanner imageSrc="foo" altText={altText} />)
    expect(
      wrapper
        .find('img')
        .props()
        .alt)
      .toBe(altText)
  })

  it('renders the titleText from props to BannerTitle ', () => {
    const titleText = 'Here is a test title!'
    const wrapper = shallow(<ImageBanner titleText={titleText} />)
    expect(wrapper).toMatchSnapshot()
    expect(
      wrapper
        .dive()
        .find(BannerTitle)
        .props().children
    ).toBe(titleText)
  })

  it('renders the subtitleText from props to BannerTitle ', () => {
    const subtitleText = 'And here is a test subtitle!'
    const wrapper = shallow(<ImageBanner subtitleText={subtitleText} />)
    expect(wrapper).toMatchSnapshot()
    expect(
      wrapper
        .dive()
        .find(BannerSubtitle)
        .props().children
    ).toBe(subtitleText)
  })
})
