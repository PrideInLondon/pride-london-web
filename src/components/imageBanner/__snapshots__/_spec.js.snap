// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBanner renders 1`] = `
<imageBanner__StyledWrapper
  allowContentUnderflow={false}
  color=""
  imageFullWidth={false}
  imageSrc=""
  large={false}
  role="banner"
>
  <imageBanner__StyledContainer
    homepage={false}
    imageFullWidth={false}
    imageSrc=""
    mx="auto"
  >
    <imageBanner__StyledRow>
      <Column
        width={1}
      >
        <StyledBannerTitle />
        <StyledBannerSubtitle />
      </Column>
    </imageBanner__StyledRow>
  </imageBanner__StyledContainer>
</imageBanner__StyledWrapper>
`;

exports[`ImageBanner renders the subtitleText from props to BannerSubtitle 1`] = `
<imageBanner__StyledWrapper
  allowContentUnderflow={false}
  color=""
  imageFullWidth={false}
  imageSrc=""
  large={false}
  role="banner"
>
  <imageBanner__StyledContainer
    homepage={false}
    imageFullWidth={false}
    imageSrc=""
    mx="auto"
  >
    <imageBanner__StyledRow>
      <Column
        width={1}
      >
        <StyledBannerTitle />
        <StyledBannerSubtitle>
          And here is a test subtitle!
        </StyledBannerSubtitle>
      </Column>
    </imageBanner__StyledRow>
  </imageBanner__StyledContainer>
</imageBanner__StyledWrapper>
`;

exports[`ImageBanner renders the titleText from props to BannerTitle  1`] = `
<imageBanner__StyledWrapper
  allowContentUnderflow={false}
  color=""
  imageFullWidth={false}
  imageSrc=""
  large={false}
  role="banner"
>
  <imageBanner__StyledContainer
    homepage={false}
    imageFullWidth={false}
    imageSrc=""
    mx="auto"
  >
    <imageBanner__StyledRow>
      <Column
        width={1}
      >
        <StyledBannerTitle>
          Here is a test title!
        </StyledBannerTitle>
        <StyledBannerSubtitle />
      </Column>
    </imageBanner__StyledRow>
  </imageBanner__StyledContainer>
</imageBanner__StyledWrapper>
`;
